common/namer/namer.go

```diff
@@ -55,3 +60,11 @@ func (n Namer) DisplayName(parts ...pulumi.StringInput) pulumi.StringInput {
 		return n.ResourceName(strArgs...)
 	}).(pulumi.StringOutput)
 }
+
+// DisplayNameWithMaxLen return pulumi.StringInput the concatanation of pulumi.StringInput `parts` prefixed
+// with the namer prefix
+func (n Namer) DisplayNameWithMaxLen(maxLen int, parts ...pulumi.StringInput) pulumi.StringInput {
+	return n.DisplayName(parts...).ToStringOutput().ApplyT(func(s string) string {
+		return utils.StrUniqueWithMaxLen(s, maxLen)
+	}).(pulumi.StringOutput)
```

Comment:
This function is:
* first concatenating the stack name with its arguments (resource name) and then
* truncate the result if it’s too long.

As a consequence, if the stack name is super verbose and the resource names very concise, the whole resource name will be systematically fully truncated whereas it would have been much more valuable to truncate the verbose stack name.

Ex.:
| `n.Ctx.Stack()` | `parts` | `maxLen` | Output of `DisplayNameWithMaxLen(maxLen, parts)` | What I would have liked |
|--------|--------|--------|--------|--------|
| `ci-17317712-4670-eks-cluster` | `fakeintake` | 32 | `ci-17317712-4670-eks-cluster-880` | `ci-17317712-4670-eks-fakeint-880` |
| `ci-17317712-4670-eks-cluster` | `nginx` | 32 | `ci-17317712-4670-eks-cluster-877` | `ci-17317712-4670-eks-cl-ngin-877` |
| `ci-17317712-4670-eks-cluster` | `redis` | 32 | `ci-17317712-4670-eks-cluster-7d0` | `ci-17317712-4670-eks-cl-redi-7d0` |

Reply:
What if the namer had more knowledge of parts ? What if we had a "StackNamer" maybe ? We could have it knowing about "stackname", resource name and user name.